lib/net/ssh/multi/session.rb

```diff
@@ -447,7 +450,7 @@ def io_select_wait(wait)
 
     def keepalive_interval
       servers = server_list.select { |s| s.options[:keepalive] }
-      servers.map { |s| s.options[:keepalive_interval] }.compact.min
+      servers.map { |s| s.options[:keepalive_interval] || DEFAULT_IO_SELECT_TIMEOUT }.compact.min
```

Comment:
I see, i thought that server list is a list of net-ssh sessions, otherwise the current options will not contains the keepalive options read by net-ssh via config files.

so this would be my modified suggestion.
```ruby
server_list.map(&:max_select_wait_time).compact.min
...

#in Net::SSH::Multi::Server:
def max_select_wait_time
  session.max_select_wait_time if session
end
```

As for testing we're already using mocha, so we should be able to do what we'd like.

Like 
```
s1 = @session.use('h1')
s1.expects(:max_select_wait).returns(20)
```

I do prefer ```max_select_wait_time``` vs ```keepalive``` as that means the two projects are more decoupled so we can change/reorg the keepalive in net-ssh shall we need to.